gh-412 - Add user friendly dialog if submission of data specification fails.

abwilson23 opened the pull request:
- Add `SimpleDialogComponent` for generic dialog messages. 
- Add a test checking the dialog is opened if an error is thrown during a test. 

Simplest way to test is to manually throw an error in one of the existing steps. I suggest a later step just to see how the UX flows from working to failure. 

closes #412

abwilson23 committed:
gh-412 - improve error messages during data spec submission.

- Log the full error to the console.
- Show a user friendly message in a simple dialog.

NigelPalmer commented on src/app/data-explorer/specification-submission/services/specification-submission.service.ts:
```diff
@@ -116,12 +115,17 @@ export class SpecificationSubmissionService {
+  private buildErrorMessage(stepName: StepName): string {
+    return `<p>Submission Step <b>${stepName}</b> failed.</p>
+    <p>Please press the Submit button again to submit your data specification.
+    If you keep seeing this message then please contact Mauro administrators for help.</p>`;
```
We are no longer showing a useful error message to the user. I just tried to submit and the user facing message told me to simply try again. Where as the console message said "NoProjectsFoundError: You do not have any projects to submit this data request for." What we need to do is display the errors that we know are user friendly, otherwise default to the current behaviour of "Please try again". I.e `if NoProjectsFoundError then display error message in dialog`

It should look something like this:
![image](https://github.com/user-attachments/assets/7f2cee0f-2448-4a5f-a535-ebd0c8b24d2a)

You may need to update the returned error message to include the additional text. Also there may be other exception types that we should be displaying the error message for.

abwilson23 replied:
NoProjectsFoundError now produces a friendly message explaining how the user can action the issue on their own. All other errors default to a standard error message from the original ticket as the fixes are outside the control of the user. 
Simple to add further modifications to error handling if necessary.

abwilson23 committed:
gh-412 - enhance error handling.

- Allow custom error handling for the NoProjectsFoundError as this is an error that the user can action themselves.

NigelPalmer committed:
gh-412: Improved submission status UX

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/MauroDataMapper/mdm-explorer?pullRequest=415) <br/>All committers have signed the CLA.

abwilson23 committed:
gh-412 - add stub method to fix failing tests.

NigelPalmer reviewed (approved):
User receives a clear message telling them what they need to do. We will likely need to update the wording in the future as the workflow changes, but the message we have now is accurate for the current state of the system.
